### ISSUE TITLE
Feature request: Statistics ?

### ISSUE THREAD

**Issue by amadeous** [ltguillaume/droidshows#88]
Hi again ;)

Thank you again for the great app! Still loving it so far!

It could be useful to have (and display) the duration of an episode (for each episode individually or on average for a serie). I have the feeling that it would "only" be a data to get from the API, a field per episode/serie to be added in the DB and to display.

Motivations:
- I though about eventually adding a statistics zone about watched series, time spent, moment of year/month/week/day episodes are watched,... Having the time of episodes of a serie would allow to compute the total time spent watching episodes/serie per day/week/month on average,... In this case, 
having the duration of each episode (instead of an average for the serie) would be more accurate.
- It could allow another feature: when several episodes are marked as seen in a row (sometimes we just can't avoid it, let's face it), the seen timestamp of the kth episode marked as seen could be incremented of k times the duration of an episode. Differently said: having the duration of an episode could allow to shift the timestamp of an episode when several are marked as seen in a row. For instance, if you add multiple episode in the end of your evening, it could add the duration of an episode to the timestamp of each episode marked as seen.

Thank you for considering this! Best!

**Comment by ltguillaume** [comment]
Hi!

Don't know if the episode's duration is a "new" feature of TheTVDB.com, but it's never been saved into DroidShow's database (only a general/average duration per show).

Statistics is a thing I thought about before (has been requested, too, I believe), but I think it would make me very, very depressed to see the amount of minutes put into mediocre shows :P

I'll think about it.

**Comment by amadeous** [comment]
The general/average duration per show could do! The goal was to store on the app the duration for general information (when we look for a show, we can see the average duration of an episode) and for potential statistic zone.

I thought it would not be a huge investment to add this duration field and that it could bring something and allow future improvement.

I could help on this statistic part if something is started ! :smiley:

**Comment by ltguillaume** [comment]
> The general/average duration per show could do!

This information is already present in the show details, next to the clock icon.

**Comment by amadeous** [comment]
Hum. Sorry, I missed that.
I changed the title of the feature request for statistics then. Let me know if you start something! I would be glad to help!

**Comment by ltguillaume** [comment]
> Hum. Sorry, I missed that.

No problemo.

> Let me know if you start something! I would be glad to help!

Would it be a "(15h20m seen)" after the episode duration in the show details, or what did you have in mind?

E: To be honest, if ever I were still to include a bigger feature, it would be https://github.com/ltGuillaume/DroidShows/issues/42 first.

**Comment by ltguillaume** [comment]
Here's the first step of stats: per-show totals in hours & minutes on the show details page.
[DroidShows_7.11.0.zip](https://github.com/ltGuillaume/DroidShows/files/4736948/DroidShows_7.11.0.zip)

@amadeous I was right though, seeing those numbers does get me depressed 😉

**Comment by amadeous** [comment]
Hello!

Thank you for this improvement! :+1: 
Wouldn't it better to split the hours into days if more than 24? I don't think days should be splitted into month (if one should/could watch that much series) since the number of days in a month is not fixed (could be splitted into weeks though?).

I listed some time ago, some statistics I thought that could be relevant/implemented and added now some of my freshest ideas. **It is not a request to implement all these elements!** Only some ideas from my head. When I say "we", I mean that I could lend a hand :wink: 
I understand that some features would go first, no problem! I am only making suggestion.

### On the show detail page

- Total time sent watching a given show

- Total number of episode watched (and percentage), including or not the special episodes according the _dialog_include_specials_ parameter

- Average time difference between air time and seen time: for a given show (avg amongst all seen episode of the show) or could be splitted for each season of the show

- There could be a representation of the distribution of seen episode over the days of the week. Could be a bar chart in percentage or a pie chart (more intuitive but less specific since there would be 7 categories so the difference might be less pronounced).

- Representation of moment during the day that episodes are watched (morning, mid-day, afternoon, evening, night). More complicated: let the user specify these time boundaries in the settings: evening should start at 6, 7, 8 or 9pm?

- …

### For more general statistics

I also had in mind a "Statistic section" where a shows would be presented globally or compared. For these, a checkbox at the top (or in the settings) could instruct to take in account or not the shows in the archive section.

**_Global stats_**
There could be the following element:

- Total time sent watching any show

- Total number of episode watched (and percentage), including or not the special episodes according the _include_specials_ parameter

- Average time difference between air time and seen time: for a any show (avg amongst all seen episode of the show). "Most/less addictive show" 

- Average time of episode amongst all show

- There could be a representation of the distribution of all seen episode over the days of the week. Could be a bar chart in percentage or a pie chart (more intuitive but less specific since there would be 7 categories so the difference might be less pronounced) or even a more complicated representation over month, see [1].

- First and last added show

- First and last completed show (including or not the special episodes according the _include_specials_ parameter)

- Percentage of show completed (including or not the special episodes according the _include_specials_ parameter)

- Favorite genre

- …

**_Table_**
There could be a table like this one:

|   | Most | Less |
|---|---|---|
| Time spent watching | Show A |  Show B |
| Time difference between air and seen time | Show B | Show C |
| Appreciated show by IMDb users[2] | Show E |  Show D |
| … |   |   |

There could be a toggle on the uppder right of the table to switch from: "In the last 7 days", "In the last 2 weeks", "In the last month", "In the last 3 months", "Of all time".

**_Listings_**
There could be listings to have more in depth representation than extremas:

- List of shows sorted be spent time seeing them

- List of shows sorted by percentage of completion

- List of genres sorted by number of show having them

- List of shows sorted by IMDb user rating

- …

A button could switch the order (ascending/descending).

**_Profile_**

We could try to display the user a "profile" of the genre they are watching like a web or star graph with percentage amongst 4 or 5 standard genre amongst: Adventure, Horror, Action, Animation, Drama, Comedy, SF,…
It could look like the activity graph on github's user profile (categories being Code review, Issue, PR, Commits).

**_More complicated (but more fun?)_**

Every show has one or more country of production so we could display a map with disks on the country of production. The disks would contain the number of shows from the country and its size would depend on this number: the larger the number, the bigger the disk. Smaller disks would be represented above the larger ones. [_A picture is worth a thousand words_](https://wpcdn.us-east-1.vip.tn-cloud.net/www.abccolumbia.com/content/uploads/2020/03/Site_COVID19_US_v22_DAP03_18_20_hpEmbed_16x9_992.jpg) (but for the whole world, not only the US).
This could also be displayed as a list of country sorted per number of shows added, started or completed.[3]

[1] Some statistics parts/representation could be inspired of the [Loop Habit app](https://github.com/iSoron/uhabits).
[2] From the IMDb user rating
[3] Could the name or even better: a flag of the country be displayed on the show detail page

**Comment by ltguillaume** [comment]
Yeah it was just a draft: days, hours, minutes would be better, on a separate line and including seen/total episode counts, but I wanted to see if stats would "make sense" to me, without adding new strings (causing the whole translations shit show to be started up again).

Seeing the sheer amount of hours I've "lost" to mediocre or just plain bad shows is not exactly inspiring, though.

**Comment by amadeous** [comment]
I get it.. It might be a motivation for some people to move out of their couch and do something fulfilling :smiley: 
Thank you for this update though! I can help for this if you need me.
